scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -519,6 +529,128 @@ impl ReplicatorConsumer {
+        self.left_range_values = values[self.table_schema.partition_key.len()..]
+            .iter()
+            .map(|el| el.map(|x| x.clone()))
+            .collect();
+    }
+
+    async fn delete_row_range_right(
+        &self,
+        data: CDCRow<'_>,
+        right_included: bool,
+    ) -> anyhow::Result<()> {
+        if self.table_schema.clustering_key.is_empty() {
+            return Ok(());
+        }
+
+        let values_before = self.left_range_values.clone();
+
+        let (_, timestamp, values) = self.get_common_cdc_row_data(&data);
+
+        let pairs_of_values = values_before
+            .iter()
+            .zip(values[self.table_schema.partition_key.len()..].iter())
+            .collect::<Vec<(&Option<_>, &Option<_>)>>();
+
+        let first_unequal_position = pairs_of_values
+            .iter()
+            .position(|pair| *pair.0 != pair.1.map(|x| x.clone()))
+            .unwrap_or(self.table_schema.clustering_key.len());
+
+        let keys_equality_cond = ReplicatorConsumer::get_keys_iter(&self.table_schema)
+            .collect::<Vec<_>>()[..self.table_schema.partition_key.len() + first_unequal_position]
+            .iter()
+            .map(|name| format!("{} = ?", name))
+            .join(" AND ");
+
+        let mut conditions = vec![keys_equality_cond];
+
+        let mut query_values =
+            values[..self.table_schema.partition_key.len() + first_unequal_position].to_vec();
+        let values = values[self.table_schema.partition_key.len()..].to_vec();
+
+        let less_than = if right_included { "<=" } else { "<" };
+        let greater_than = if self.left_range_included { ">=" } else { ">" };
+
+        let first_left_null_index = values_before[first_unequal_position..]
+            .iter()
+            .position(|x| x.is_none())
+            .map_or(self.table_schema.clustering_key.len(), |x| {
+                x + first_unequal_position
+            });
+        query_values.extend(
+            values_before[first_unequal_position..first_left_null_index]
+                .iter()
+                .map(|el| el.as_ref()),
+        );
+
+        match first_left_null_index.cmp(&(first_unequal_position + 1)) {
+            Ordering::Less => {}
+            Ordering::Equal => {
+                conditions.push(format!(
+                    "{} {} ?",
+                    self.table_schema.clustering_key[first_unequal_position], greater_than
+                ));
+            }
+            Ordering::Greater => {
+                conditions.push(format!(
+                    "({}) {} ({})",
+                    self.table_schema.clustering_key[first_unequal_position..first_left_null_index]
+                        .iter()
+                        .join(","),
+                    greater_than,
+                    &"?,".repeat(first_left_null_index - first_unequal_position)
+                        [..2 * (first_left_null_index - first_unequal_position) - 1]
```

Comment:
The use of slice here is confusing - I suppose its purpose was to skip the last comma. Why not use join like before? Won't something like this work instead?

```rust
std::iter::repeat("?").take(first_left_null_index - first_unequal_position).join(",")
```